src/main.ts

```diff
@@ -82,9 +87,111 @@ function createWindow() {
   ipcMain.on('doScreenshot', handleDoScreenshot)
   ipcMain.handle('getVersion', handleGetVersion)
 
+  ipcMain.handle('prepareWebsite', handlePrepareWebsite)
+  ipcMain.handle('displayWebsite', handleDisplayWebsite)
+  ipcMain.handle('destroyWebsite', handleDestroyWebsite)
+  ipcMain.handle('prepareWebsiteFullscreen', handlePrepareWebsiteFullscreen)
+  ipcMain.handle('displayWebsiteFullscreen', handleDisplayWebsiteFullscreen)
+  ipcMain.handle('destroyWebsiteFullscreen', handleDestroyWebsiteFullscreen)
+
   autoUpdater.checkForUpdatesAndNotify()
 }
 
+let preparingFullscreenWebsiteId = ''
+let displayingFullscreenWebsiteId = ''
+let displayWebsites: { [keys: string]: Electron.BrowserWindow } = {}
+function handlePrepareWebsite(
+  event: any,
+  url: string,
+  code: string,
+  orientation: number,
+  x: number,
+  y: number,
+  width: number,
+  height: number,
+  id: string
+) {
+  handlePrepareWebsiteFullscreen(event, url, code, orientation)
+}
+
+function handleDisplayWebsite(event: any, id: string) {
+  handleDisplayWebsiteFullscreen(event)
+}
+
+function handleDestroyWebsite(event: any, id: string) {
+  handleDestroyWebsiteFullscreen(event)
+}
+
+function handlePrepareWebsiteFullscreen(
+  _event: any,
+  url: string,
+  code: string,
+  orientation: number
+) {
+  console.log('handlePrepareWebsiteFullscreen')
+  if (preparingFullscreenWebsiteId) {
+    handleDestroyWebsiteFullscreen(_event, preparingFullscreenWebsiteId)
+  }
+
+  const id = Math.random().toString()
+  preparingFullscreenWebsiteId = id
+  console.log('preparing ' + id)
+
+  displayWebsites[id] = new BrowserWindow({
+    webPreferences: {
+      preload: path.join(__dirname, 'preload.js'),
+      webSecurity: false,
+      allowRunningInsecureContent: true,
+      autoplayPolicy: 'no-user-gesture-required',
+      nodeIntegration: false,
+    },
+    autoHideMenuBar: true,
+    kiosk: app.isPackaged,
+    alwaysOnTop: false,
+    show: false,
+    fullscreen: true,
+  })
+  displayWebsites[id].loadURL(url)
+
+  displayWebsites[id].webContents.on('did-finish-load', () => {
+    console.log('executeJavaScript')
+    displayWebsites[id].webContents.executeJavaScript(code)
+  })
+}
+
+function handleDisplayWebsiteFullscreen(_event: any) {
+  console.log('handleDisplayWebsiteFullscreen' + preparingFullscreenWebsiteId)
+  displayingFullscreenWebsiteId = preparingFullscreenWebsiteId
+  preparingFullscreenWebsiteId = ''
+  mainWindow.setAlwaysOnTop(false)
+  mainWindow.hide()
```

Comment:
have to hide the main window because it was covering up the displayWebsite window. The main window is then restored in handleDestroyWebsiteFullscreen function